src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2991,7 +2991,7 @@ private void OnProjectStarted(object sender, ProjectStartedEventArgs e)
 
                 loggers = (loggers ?? Enumerable.Empty<ILogger>()).Concat(new[]
                 {
-                    new BuildCheckConnectorLogger(new AnalyzerLoggingContextFactory(loggingService), buildCheckManagerProvider.Instance)
+                    new BuildCheckConnectorLogger(new AnalyzerLoggingContextFactory(loggingService), buildCheckManagerProvider.Instance, _buildParameters.AreBuildCheckStatsEnabled)
```

Comment:
How expensive is the process of this collection? Could we just have it on all the time?

Reply:
Oh, it actually is and this controls only log verbosity? I . . . am not sure that's worth the complexity. Is there a strong need for it?

Reply:
It currently it is a switch for verbosity. It was originally for collecting this infrastructure data in general, but I was asked to change that after a discussion that the stats would be part of the analyzers (https://github.com/dotnet/msbuild/pull/9890#discussion_r1560706854)